lerobot/teleoperate.py

```diff
@@ -0,0 +1,133 @@
+
+Example:
+
+```shell
+python -m lerobot.teleoperate \
+    --robot.type=so101_follower \
+    --robot.port=/dev/tty.usbmodem58760431541 \
+    --robot.cameras="{ front: {type: opencv, index_or_path: 0, width: 1920, height: 1080, fps: 30}}" \
+    --robot.id=black \
+    --teleop.type=so101_leader \
+    --teleop.port=/dev/tty.usbmodem58760431551 \
+    --teleop.id=blue \
+    --display_data=true
+```
+"""
+
+import logging
+import time
+from dataclasses import asdict, dataclass
+from pprint import pformat
+
+import draccus
+import numpy as np
+import rerun as rr
+
+from lerobot.common.cameras.opencv.configuration_opencv import OpenCVCameraConfig  # noqa: F401
+from lerobot.common.cameras.realsense.configuration_realsense import RealSenseCameraConfig  # noqa: F401
+from lerobot.common.robots import (  # noqa: F401
+    Robot,
+    RobotConfig,
+    koch_follower,
+    make_robot_from_config,
+    so100_follower,
+    so101_follower,
+)
+from lerobot.common.teleoperators import (
+    Teleoperator,
+    TeleoperatorConfig,
+    make_teleoperator_from_config,
+)
+from lerobot.common.utils.utils import init_logging, move_cursor_up
+from lerobot.common.utils.visualization_utils import _init_rerun
+
+from .common.teleoperators import koch_leader, so100_leader, so101_leader  # noqa: F401
+
+
+@dataclass
+class TeleoperateConfig:
+    teleop: TeleoperatorConfig
+    robot: RobotConfig
+    # Limit the maximum frames per second. By default, no limit.
+    fps: int | None = None
+    teleop_time_s: float | None = None
+    # Display all cameras on screen
+    display_data: bool = False
+
+
+def teleop_loop(
+    teleop: Teleoperator, robot: Robot, display_data: bool = False, duration: float | None = None
+):
+    display_len = max(len(key) for key in robot.action_features)
+    start = time.perf_counter()
+    while True:
+        loop_start = time.perf_counter()
+        action = teleop.get_action()
+        if display_data:
+            observation = robot.get_observation()
+            for obs, val in observation.items():
+                if isinstance(val, float):
+                    rr.log(f"observation_{obs}", rr.Scalar(val))
+                elif isinstance(val, np.ndarray):
+                    rr.log(f"observation_{obs}", rr.Image(val), static=True)
+            for act, val in action.items():
+                if isinstance(val, float):
+                    rr.log(f"action_{act}", rr.Scalar(val))
+
+        robot.send_action(action)
+        loop_s = time.perf_counter() - loop_start
+
+        print("\n" + "-" * (display_len + 10))
+        print(f"{'NAME':<{display_len}} | {'NORM':>7}")
+        for motor, value in action.items():
+            print(f"{motor:<{display_len}} | {value:>7.2f}")
+        print(f"\ntime: {loop_s * 1e3:.2f}ms ({1 / loop_s:.0f} Hz)")
+
+        if duration is not None and time.perf_counter() - start >= duration:
+            return
+
+        move_cursor_up(len(action) + 5)
+
```

Comment:
Should we add back a `busy_wait` to respect the fps value that is not used in the current function @aliberts ?
```suggestion
        if fps is not None:
            busy_wait(1 / fps - (time.perf_counter() - loop_start))
```
I noticed teleoperating with a gamepad with an infinite frame rate is quite difficult.